packages/meditrak-server/src/tests/routes/importSurveyResponses/testOutdatedStatusUpdate.js

```diff
@@ -0,0 +1,114 @@
+  },
+};
+
+export const testOutdatedStatusUpdate = () => {
+  const app = new TestableApp();
+  const { models } = app;
+  const surveyResponseChangeHandler = new SurveyResponseChangeHandler(
+    models,
+    REFRESH_DEBOUNCE_TIME,
+  );
+
+  /**
+   * Asserts that newly imported responses for a survey have the expected `outdated` statuses.
+   * Statuses should follow the ascending id order of their corresponding survey responses:
+   * for newly created responses, this should match the column order in the import spreadsheet
+   */
+  const assertOutdatedStatuses = async (surveyCode, expectedOutdatedStatuses) => {
+    const surveyResponses = await models.surveyResponse.find(
+      { survey_id: SURVEYS[surveyCode].id },
+      { sort: ['id'] },
+    );
+    const wrongCountMessage = `Count of survey responses for survey "${surveyCode}" must match that of the provided outdated statuses`;
+    expect(surveyResponses).to.have.lengthOf(expectedOutdatedStatuses.length, wrongCountMessage);
+
+    for (let i = 0; i < surveyResponses.length; i++) {
+      const surveyResponse = surveyResponses[i];
+
+      const entity = await models.entity.findById(surveyResponse.entity_id);
+      const srDescriptionFields = {
+        survey_code: surveyCode,
+        entity_code: entity.code,
+        data_time: surveyResponse.data_time,
+      };
+      const message = `Failed assertion for survey response ${JSON.stringify(srDescriptionFields)}`;
+
+      // const message = `Outdated status for responses of survey response\n${} is different than expected`;
+      expect(surveyResponse).to.have.property('outdated', expectedOutdatedStatuses[i], message);
+    }
+  };
+
+  before(async () => {
+    await app.grantFullAccess();
+    await buildAndInsertSurveys(models, Object.values(SURVEYS));
+  });
+
+  beforeEach(async () => {
+    await models.surveyResponse.delete({ survey_id: Object.values(SURVEYS).map(s => s.id) });
+    surveyResponseChangeHandler.listenForChanges();
+  });
+
+  afterEach(async () => {
+    surveyResponseChangeHandler.stopListeningForChanges();
+  });
+
+  after(() => {
+    app.revokeAccess();
+  });
+
+  it('importing new survey responses using a mix of surveys, entities and timestamps', async () => {
+    const response = await importFile(
+      app,
+      'outdatedStatus.xlsx',
+      Object.values(SURVEYS).map(s => s.name),
+    );
+
+    expect(response.statusCode).to.equal(200);
+
+    await models.database.waitForAllChangeHandlers();
+    // Statuses listed in the same order as the corresponding survey responses in the import spreadsheet
+    await assertOutdatedStatuses('Test_Yearly', [true, true, false, false, false, false]);
+    await assertOutdatedStatuses('Test_Monthly', [true, true, false, false, false, false]);
+    // Responses for non periodic surveys are never "outdated"
+    await assertOutdatedStatuses('Test_No_Granularity', [false, false, false, false, false, false]);
```

Comment:
Just record creation tests here, since updates/deletes are tested in the corresponding `surveyResponse` routes (get/put/delete). Although given the significance of the import process in our ecosystem, it may be worth adding a few more cases here

Reply:
Would also love to see those tests added, but won't push on it 👍